src/Transformer/Normalization/BooleanTransformer.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flagbit\Bundle\CategoryBundle\Transformer\Normalization;
+
+use Flagbit\Bundle\CategoryBundle\Transformer\NormalizationTransformer;
+
+/**
+ * Transformer that handles proper normalization of boolean values.
+ */
+class BooleanTransformer implements NormalizationTransformer
+{
+    /**
+     * Convert passed data to a int representing the boolean value of the data (0 / 1).
+     *
+     * @param mixed $data
+     */
+    public function transform($data): int
+    {
+        return (int) ((bool) $data);
+    }
```

Comment:
`$data` is a boolean, return is either `'0'` or `'1'`. Checkboxes in the frontend allow only `true` or `false`.

Maybe

```suggestion
    public function transform($data)
    {
        return $data ? '1' : '0';
    }
```
would be fitting?

Also for phpstan: `@phpstan-return numeric-string`. See https://phpstan.org/writing-php-code/phpdoc-types#other-advanced-string-types